src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -189,19 +190,28 @@ public Action createAdHocAction(final Action action) {
     return actionRepo.saveAndFlush(action);
   }
 
-  @Transactional
-  public void createScheduledActions(ActionPlan actionPlan, ActionPlanJob actionPlanJob) {
-    List<ActionCase> cases = actionCaseRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
-    List<ActionRule> rules = actionRuleRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public void createScheduledActions(Integer actionPlanPk) {
+    // Action plan job has to be created before actions
+    ActionPlanJob actionPlanJob = createActionPlanJob(actionPlanPk);
+
+    List<ActionCase> cases = actionCaseRepo.findByActionPlanFK(actionPlanPk);
+    List<ActionRule> rules = actionRuleRepo.findByActionPlanFK(actionPlanPk);
+    List<ActionType> types = actionTypeRepo.findAll();
 
     // For each case/rule pair create an action
-    cases.forEach(caze -> createActionsForCase(caze, rules));
-    updatePlanAndJob(actionPlan, actionPlanJob);
+    cases.forEach(caze -> createActionsForCase(caze, rules, types));
```

Comment:
I'm aware that this is pre-existing, but I just want to flag that this is quite a computationally complex piece of code dealing with large datasets and if it's possible to break the foreach nesting (I don't think it is) we should. I appreciate that the changes to where we flush has made this a lot better :)